nbgitpuller/plugins/plugin_helper.py

```diff
@@ -0,0 +1,116 @@
+import subprocess
+import os
+import logging
+import requests
+from requests_file import FileAdapter
```

Comment:
Since `requests` is blocking, we can not use it here :( I'd suggest we use [aiohttp](https://docs.aiohttp.org/en/stable/) instead. We can use [aioresponses](https://github.com/pnuckowski/aioresponses) to mock the response in tests